pkg/audit/audit.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package audit
+
+import (
+	"bytes"
+	"context"
+	"crypto/ecdsa"
+	"io"
+
+	"github.com/vivint/infectious"
+	"google.golang.org/grpc"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/piecestore/rpc/client"
+	"storj.io/storj/pkg/provider"
+)
+
+func setupPSClient(ctx context.Context) (psClient client.PSClient, err error) {
+	// TODO(nat): get this info from the satellite somehow
+	ca, err := provider.NewCA(ctx, 12, 4)
+	if err != nil {
+		return nil, err
+	}
+	identity, err := ca.NewIdentity()
+	if err != nil {
+		return nil, err
+	}
+	identOpt, err := identity.DialOption()
+	if err != nil {
+		return nil, err
+	}
+
+	var conn *grpc.ClientConn
+	conn, err = grpc.Dial(":7777", identOpt)
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+
+	psClient, err = client.NewPSClient(conn, 1024*32, identity.Key.(*ecdsa.PrivateKey))
+	if err != nil {
+		return nil, err
+	}
+	return psClient, nil
+}
```

Comment:
I'm not sure about setting up the psclient in the audit library -  I personally left that out and just instantiated it with . a new  instance of audit: `&audit{psclient:psclient}`.  I don't think it's the library job to setup  a psclient.